sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -232,7 +232,7 @@ class Analyzer(
         // substitute the group by expressions.
         val newGroupByExprs = groupByExprPairs.map(_._2)
```

Comment:
Take this for example:
```sql
select sum(a-b) as ab from mytable group by b with rollup;
```
I think we probably need to add extra column(s) for the output of `Expand`, (e.g. `(a-b)` in this case). Will that be more simple for this fixing?